kadmin/server.c

```diff
@@ -341,13 +372,12 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
 		ret = KADM5_PASS_Q_DICT;
 	    else
 		ret = 0;
-	} else
-	    ret = _kadm5_acl_check_permission(contextp, KADM5_PRIV_CPW, princ);
-
-	if(ret) {
-	    krb5_free_principal(contextp->context, princ);
-	    goto fail;
+	    if (ret) {
+		krb5_free_principal(contextp->context, princ);
```

Comment:
Note to self: we should move this very-common-in-this-function cleanup into the `fail` processing.